lib/common/ipc.c

```diff
@@ -579,20 +579,18 @@ crm_ipc_prepare(uint32_t request, xmlNode * message, struct iovec ** result, uin
 
             free(buffer);
 
-            if (header->size_compressed > biggest) {
-                biggest = 2 * QB_MAX(header->size_compressed, biggest);
-            }
+            biggest = QB_MAX(header->size_compressed, biggest);
 
         } else {
             ssize_t rc = -EMSGSIZE;
 
             crm_log_xml_trace(message, "EMSGSIZE");
-            biggest = 2 * QB_MAX(header->size_uncompressed, biggest);
+            biggest = QB_MAX(header->size_uncompressed, biggest);
```

Comment:
Sorry, just got back from vacation :)

As long as the following message cannot fit into the current buffer, even if it's only 1 byte larger, it'll warn again anyway, right?

It'd also work if you prefer to move the factor '4' from the logging line into here. Then "biggest" would basically record 4 times of the real biggest size ;-)